bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqBinding.java

```diff
@@ -0,0 +1,381 @@
+		if (StringUtils.isNotBlank(usernameString)
+				&& StringUtils.isNotBlank(passwordString)) {
+			myqOnlineData = new MyqData(usernameString, passwordString);
+		}
+		invalidCredentials = false;
+		schedulePoll(refreshInterval);
+	}
+
+	/**
+	 * Called by the SCR to deactivate the component when either the
+	 * configuration is removed or mandatory references are no longer satisfied
+	 * or the component has simply been stopped.
+	 * 
+	 * @param reason
+	 *            Reason code for the deactivation:<br>
+	 *            <ul>
+	 *            <li>0 – Unspecified
+	 *            <li>1 – The component was disabled
+	 *            <li>2 – A reference became unsatisfied
+	 *            <li>3 – A configuration was changed
+	 *            <li>4 – A configuration was deleted
+	 *            <li>5 – The component was disposed
+	 *            <li>6 – The bundle was stopped
+	 *            </ul>
+	 */
+	public void deactivate(final int reason) {
+		this.bundleContext = null;
+		// deallocate resources here that are no longer needed and
+		// should be reset when activating this binding again
+
+		if (pollFuture != null && !pollFuture.isCancelled())
+			pollFuture.cancel(true);
+
+		if (pollResetFuture != null && !pollResetFuture.isCancelled())
+			pollResetFuture.cancel(true);
+	}
+
+	/**
+	 * Poll for device changes
+	 */
+	private void poll() {
+		if (invalidCredentials || this.myqOnlineData == null) {
+			logger.debug("Invalid Account Credentials");
+			return;
+		}
+
+		try {
+			// Get myQ Data
+			GarageDoorData garageStatus = myqOnlineData.getGarageData();
+
+			for (MyqBindingProvider provider : providers) {
+				for (String mygItemName : provider.getInBindingItemNames()) {
+					MyqBindingConfig deviceConfig = getConfigForItemName(mygItemName);
+
+					if (deviceConfig != null) {
+						GarageDoorDevice garageopener = garageStatus
+								.getDevice(deviceConfig.deviceIndex);
+						if (garageopener != null) {
+
+							State newState = UnDefType.UNDEF;
+							for (Class<? extends State> type : deviceConfig.acceptedDataTypes) {
+								if (StringType.class == type) {
+									newState = new StringType(garageopener
+											.getStatus().getLabel());
+									break;
+								} else if (OpenClosedType.class == type) {
+									if (garageopener.getStatus() == GarageDoorStatus.CLOSED) {
+										newState = OpenClosedType.CLOSED;
+										break;
+									} else {
+										newState = OpenClosedType.OPEN;
+										break;
+									}
+								} else if (UpDownType.class == type) {
+									if (garageopener.getStatus() == GarageDoorStatus.CLOSED) {
+										newState = UpDownType.DOWN;
+										break;
+									} else if (garageopener.getStatus() == GarageDoorStatus.OPEN) {
+										newState = UpDownType.UP;
+										break;
+									}
+								} else if (PercentType.class == type) {
+									if (garageopener.getStatus() != GarageDoorStatus.UNKNOWN) {
+										newState = new PercentType(50);
```

Comment:
We know when the door is in motion or is stuck in in a half open state, we don't know exactly what percentage the door is open or closed. Roller shutter items afaik only support OPEN/CLOSE and percentage.  I put this there so that roller shutter items could display the correct state icon, so in my sitemap the  icon goes from close -> halfway open -> open.